internal/pkg/deploy/cloudformation/stack/transformers.go

```diff
@@ -66,22 +66,45 @@ var (
 	}
 )
 
+// findContainerNameGivenPort searches through exposed ports in the manifest to find the container name that matches the given port.
+func findContainerNameGivenPort(port uint16, exposedPorts []manifest.ExposedPort) string {
+	for _, portConfig := range exposedPorts {
+		if portConfig.Port == port {
+			return portConfig.ContainerName
+		}
+	}
+	return ""
+}
+
+func convertPortMapping(exposedPorts []manifest.ExposedPort) map[string][]*template.PortMapping {
+	portMapping := make(map[string][]*template.PortMapping)
+	for _, exposedPort := range exposedPorts {
+		out := &template.PortMapping{
+			ContainerPort: exposedPort.Port,
+			Protocol:      exposedPort.Protocol,
+			ContainerName: exposedPort.ContainerName,
+		}
+		portMapping[exposedPort.ContainerName] = append(portMapping[exposedPort.ContainerName], out)
+	}
+	return portMapping
+}
+
 // convertSidecar converts the manifest sidecar configuration into a format parsable by the templates pkg.
-func convertSidecar(s map[string]*manifest.SidecarConfig) ([]*template.SidecarOpts, error) {
-	if s == nil {
+func convertSidecar(sidecarsMap map[string]*manifest.SidecarConfig, portMappings map[string][]*template.PortMapping) ([]*template.SidecarOpts, error) {
```

Comment:
```suggestion
func convertSidecar(sidecarsMap map[string]*manifest.SidecarConfig, portMappings []*template.PortMapping) ([]*template.SidecarOpts, error) {
```

Reply:
portmappings is in the form of `map[constianerName][]template.PortMapping` which helps us easily set particular container's portmappings to its config.